shopfloor/services/location_content_transfer.py

```diff
@@ -0,0 +1,921 @@
+                    "product_uom": quant.product_uom_id.id,
+                    "product_uom_qty": quant.quantity,
+                    "location_id": location.id,
+                    "location_dest_id": picking_type.default_location_dest_id.id,
+                    "origin": self.work.menu.name,
+                    "picking_type_id": picking_type.id,
+                }
+            )
+        return self.env["stock.move"].create(move_vals_list)
+
+    def scan_location(self, barcode):
+        """Scan start location
+
+        Called at the beginning at the workflow to select the location from which
+        we want to move the content.
+
+        All the move lines and package levels must have the same picking type.
+
+        If the scanned location has no move lines, new move lines to move the
+        whole content of the location are created if:
+
+        * the menu has the option "Allow to create move(s)"
+        * the menu is linked to only one picking type.
+
+        When move lines and package levels have different destinations, the
+        first line without package level or package level is sent to the client.
+
+        Transitions:
+        * start: location not found, ...
+        * scan_destination_all: if the destination of all the lines and package
+        levels have the same destination
+        * start_single: if any line or package level has a different destination
+        """
+        location = self.actions_for("search").location_from_scan(barcode)
+        if not location:
+            return self._response_for_start(message=self.msg_store.barcode_not_found())
+        move_lines = self._find_location_move_lines(location)
+        pickings = move_lines.mapped("picking_id")
+        picking_types = pickings.mapped("picking_type_id")
+
+        if len(picking_types) > 1:
+            return self._response_for_start(
+                message={
+                    "message_type": "error",
+                    "body": _("This location content can't be moved at once."),
+                }
+            )
+        if picking_types - self.picking_types:
+            return self._response_for_start(
+                message={
+                    "message_type": "error",
+                    "body": _("This location content can't be moved using this menu."),
+                }
+            )
+        # If the following criteria are met:
+        #   - no move lines have been found
+        #   - the menu is configured to allow the creation of moves
+        #   - the menu is bind to one picking type
+        # then prepare new stock moves to move goods from the scanned location.
+        menu = self.work.menu
+        if (
+            not move_lines
+            and menu.allow_move_create
+            and len(menu.picking_type_ids) == 1
+        ):
+            new_moves = self._create_moves_from_location(location)
+            new_moves._action_confirm(merge=False)
+            new_moves._action_assign()
```

Comment:
seems like we can have an issue here, in case we can't assign (goods reserved already?), we won't be able to move the content or only partially, maybe we should rollback the creation and return an error message?

Reply:
fixed by @simahawk in sebalix#1